www/js/app.js

```diff
@@ -837,8 +837,14 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
                         return false;
                     });
                 }
-                else if (url.slice(0, 1) === "#") {
-                    // It's an anchor link : do nothing
+                else if (util.startsWith($(this).attr("href"), "#")) {
+                    // It's an anchor link : we need to make it work with javascript
+                    // because of the base tag
+                    var anchor = $(this).attr("href");
+                    $(this).on('click', function(e) {
+                        $('#articleContent').first()[0].contentWindow.location.hash = anchor;
+                        return false;
+                    });
```

Comment:
@mossroy So, I've tested a slight simplification of your approach, and I think substituting this for the above else if clause would do the same job, without requiring the utility function:
```
                else if (this.hash && ~this.href.indexOf("/#")) {
                    // It's an anchor link : we need to make it work with javascript
                    // because of the base tag
                    var anchor = this.hash;
                    $(this).on('click', function (e) {
                        $('#articleContent').first()[0].contentWindow.location.hash = anchor;
                        return false;
                    });
                }
```
What do you think? If you prefer your approach for reasons of clarity, that's completely fine by me.

Reply:
It's not clear to me of what you're trying to test with `~this.href.indexOf("/#")`
I'm not used to this tilde operator, but I understood that, in this context, it is true if "/#" is found in this.href, and false if it is not found.
Why are you looking for "/#"?

Reply:
The ``~this.href.indexOf("/#")`` ensures that the anchor is a local one, anticipating your previous comment that ``this.hash`` on its own could match an anchor on an external site.

A perhaps less elegant alternative could be:

``else if (/\/#[^/]+$/.test(this.href)) {`` 

(no need for ``this.hash`` in this case).

Reply:
I don't understand why the string "/#" would indicate a local anchor. If I there is in the article a hyperlink to http://www.geeksphone.com/#our-passion for example (this is a real URL), it will match even if it is not a local anchor.

Reply:
Ah, OK, I hadn't thought of such URLs, as none of the pages I tested had that form -- silly me for trying to get around using jQuery...

So, if all you want to do is test if the URL begins with a hash, then ` else if (/^#/.test($(this).attr("href")))` will do the job. Or store it in a variable earlier and test for both "" and # in the same condition.

Reply:
I agree with you, and will push an updated fix during the weekend.
I'll probably refactor a bit this part of code, which should hopefully fix #313 too